### PR TITLE
fix: rebuild index when clear storage

### PR DESCRIPTION
**Pull request by AnneYang720**
Signed-off-by: AnneY <evangeline-lun@foxmail.com>

Goals:

This PR is related to issue #829  and will rebuild index for Redis and ElasticSearch when `_clear_storage` is called.

- [x] change for Redis
- [x] change for ElasticSearch
- [x] add related test

### CODE REVIEW

File: docarray/array/storage/elastic/backend.py [hunk 1]
```diff
@@ -166,22 +172,15 @@ def _build_schema_from_elastic_config(self, elastic_config):
             ] = index_options
         return da_schema
 
-    def _build_client(self):
```
Comment (JohannesMessner) [inline docarray/array/storage/elastic/backend.py:169]: I think we can keep this method and do `self._client = self._build_client()`, while still keeping `_build_index()`? That way we don't need to repeat these lines down in line 289

File: docarray/array/storage/redis/backend.py [hunk 2]
```diff
@@ -84,7 +84,14 @@ def _init_storage(
         self._doc_prefix = config.index_name + ':'
         self._config.columns = self._normalize_columns(self._config.columns)
 
-        self._client = self._build_client()
+        self._client = Redis(
```
Comment (JohannesMessner) [inline docarray/array/storage/redis/backend.py:87]: same here, i think we can keep the _buid_client() method

File: docarray/array/storage/redis/backend.py [hunk 3]
```diff
@@ -94,25 +101,19 @@ def _init_storage(
...
-        client = Redis(
-            host=self._config.host,
-            port=self._config.port,
-            **self._config.redis_config,
-        )
-
-        if self._config.update_schema:
-            if self._config.index_name.encode() in client.execute_command('FT._LIST'):
-                client.ft(index_name=self._config.index_name).dropindex()
+    def _build_index(self, rebuild: bool = False):
```
Comment (JohannesMessner) [inline docarray/array/storage/redis/backend.py:104]: I am not familiar with the inner workings of redis, could you very briefly explain why this `rebuild` argument is needed?

Comment (AnneYang720) [reply docarray/array/storage/redis/backend.py:104]: RedisConfig has boolean `update_schema` to decide whether to update Redis Search index schema during initialization. I want the function `_build_index` to be used by both `_init_storage` and rebuild index of `_clear_storage`. This `rebuild` argument is for the latter situation.

**Review by JohannesMessner — APPROVED** [review]

**Review by JoanFM — APPROVED** [review]

**Pull request merged**